airflow/jobs/scheduler_job.py

```diff
@@ -1409,6 +1399,15 @@ def _run_scheduler_loop(self) -> None:
                 # usage when "idle"
                 time.sleep(self._processor_poll_interval)
 
+            # Verify that all files were processed, and if so, deactivate DAGs that
+            # haven't been touched by the scheduler as they likely have been deleted.
+            if self.processor_agent.all_files_processed:
```

Comment:
@ashb do we still need to keep this check? I feel that we don't need to keep it anymore.

Reply:
No, not needed anymore, correct.